docs/epm/index.asciidoc

```diff
@@ -31,3 +31,39 @@ Delete a package:
 curl --user elastic:changeme localhost:5601/api/epm/delete/iptables-1.0.4
 curl --user elastic:changeme localhost:5601/api/epm/delete/coredns-1.0.1
 ```
+
+== Indexing Strategy
+
+Ingest Management enforces an indexing strategy to allow the system to automically detect indices and run queries on it. In short the indexing strategy looks as following:
+
+```
+{type}-{namespace}-{dataset}
+```
+
+The `{type}` can be `logs` or `metrics`. The `{namespace}` is the part where the user can use free form. The only requirement is that it is valid in an Elasticsearch index and does NOT contain a `-`. The `dataset` is defined by the data that is indexed. A `dataset` must also be valid in an Elasticsearch index and not contain `-`. It is expected that the fields for type, namespace and dataset are part of each event and are constant keywords.
+
+Note: More `{type}`s might be added in the future like `apm` and `endpoint`.
```

Comment:
I wonder how do we decide on when we create a new `{type}` and when to reuse `logs`. There's a large number of events that feel like logs, but are not logs. Like alerts, network flows, process executions, packetbeat transactions, etc.  ECS's [event.kind](https://www.elastic.co/guide/en/ecs/1.4/ecs-allowed-values-event-kind.html) currently lists 6 values. Maybe there's an opportunity to reuse some of that?

Reply:
Good question. So far the focus was much more on what "types" do we start with and with filebeat and metricbeat the two most obvious were logs and metrics. In general I would like to keep the types to a minimum as Elasticsearch will have to ship with a basic template for these types to make sure everything works out of the box.

The definition of the index should rely on ECS fields. It would have been nice to use `{event.type}-{event.namespace}-{event.dataset}`. Unfortunately `event.type` has already a different meaning. It seems what `event.type` is should actually be `event.category.type` ;-)

For `event.kind`: I think it is close but not the same. The value that suprised me at first is `pipeline_error` as having an error in a pipeline still means the event belongs to the same dataset. It seems events from 1 log file can have different `event.kind` values.

I think there is also an opportunity for some apps to create their own type like using `signal-*-*` for SIEM as an example. As far as I understand, this index would only be created when SIEM is used for the first time and could be set up by the SIEM app?

Getting back to the initial question: How do we decide on the types? Lets turn around the question: What are the types you would need?

Reply:
Side note: Thinking about which fields this could fit in: `stream.type`, `event.namespace`, `event.dataset`. The reason for the `stream.*` is that it is what we currently plan to use as the config on the agent side.